docs/docs/nlu-training-data.mdx

```diff
@@ -3,92 +3,166 @@ id: nlu-training-data
+
+If your entity has a deterministic structure, you can use regular expressions in one of two ways:
+
+1. As features for the [`RegexFeaturizer`](components.mdx#regexfeaturizer) component in the pipeline.
 
-### Entities Roles and Groups
+  When used as features for the `RegexFeaturizer` the
+  name of the regular expression does not matter.
+  When using the `RegexFeaturizer`, a regular expression provides a feature
+  that helps the model learn an association between intents/entities and inputs
+  that fit the regular expression. 
+  
+  :::note Needs Training Examples
+  The `RegexFeaturizer` provides features to the entity extractor, but it doesn't predict the entity directly. Include enough examples containing the regular expression so that the entity extractor can learn ot use the regular expression feature.
+
+  :::
+
+  Regex features for entity extraction
+  are currently only supported by the `CRFEntityExtractor` and `DIETClassifier` components. Other entity extractors, like
+  `MitieEntityExtractor` or `SpacyEntityExtractor`, won't use the generated
+  features and their presence will not improve entity recognition for
+  these extractors.
+
+2. For rule-based entity extraction using the [`RegexEntityExtractor`](components.mdx#regexentityextractor) component in the pipeline.
+
+  When using the `RegexEntityExtractor`, the name of the regular expression should
+  match the name of the entity you want to extract.
+  For example, you could extract account numbers of 10-12 digits by including this regular expression and at least two annotated examples in your training data:
+
+  ```yaml-rasa
+  nlu:
+  - regex: account_number
+    examples: |
+      - \d{10,12}
+  - intent: inform
+    examples: |
+      - my account number is [1234567891](account_number)
+      - This is my account number [1234567891](account_number)
+  ```
+
+  Whenever a user message contains a sequence of 10-12 digits, it will be extracted as an `account_number` entity. `RegexEntityExtractor` doesn't require training examples to learn to extract the entity, but you do need at least two annotated examples of the entity so that the NLU model can register it as an entity at training time.
+
+
+## Lookup Tables
+
+Lookup tables are lists of words used to generate
+case-insensitive regular expression patterns. They can be used in the same ways as [regular expressions](#regular-expressions) are used, in combination with the [`RegexFeaturizer`](components.mdx#regexfeaturizer) and [`RegexEntityExtractor`](components.mdx#regexentityextractor) components in the pipeline.
+
+You can use lookup tables to help extract entities which have a known set of possible values. Keep your lookup tables as specific as possible. For example, to extract country names, you could add a lookup table of all countries in the world:
+
+```yaml-rasa
+nlu:
+- lookup: country
+  examples: |
+    - Afghanistan
+    - Albania
+    - ...
+    - Zambia
+    - Zimbabwe
+```
```

Comment:
can we still provide a path to a file?

Reply:
no, but the lookup could go in its own file.

Reply:
coolio, thanks